### PR TITLE
4711 Update colin api dependencies required to resolve data loader conflicts

### PR DESCRIPTION
**Pull request by argush3**
*Issue #:* /bcgov/entity#4711

Note: this is an initial PR to resolve dependency issues to get data loader to a state where it can run.  As data loader, pulls in the colin api dependencies using git+https://github.com/bcgov/lear.git#egg=colin_api&subdirectory=colin-api in the requirements file, the updates in this PR need to be merged into master before data loader is able to pull in dependencies conflict free.  The actual implementation for 4711 will be coming in a subsequent PR.

*Description of changes:*

- Updated requirements files to resolve dependencies that is required to run data loader
- Updated code to use restx where restplus was previously used to resolve dependency conflict issues
- Updated .gitignore to ignore all .iml files(IDE generated artifact)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Review by kialj876 — APPROVED** [review]

File: colin-api/Makefile [hunk 1]
```diff
@@ -37,7 +37,7 @@ clean-test: ## clean test files
 	rm -fr htmlcov/
 
 build-req: clean ## Upgrade requirements
-	test -f venv/bin/activate || python3.8 -m venv  $(CURRENT_ABS_DIR)/venv ;\
```
Comment (thorwolpert) [inline colin-api/Makefile:40]: Checking with the team, but the CI maintainers want it at 3.8

**Review by thorwolpert — CHANGES_REQUESTED** [review]
The CI groups wants to keep the Makefiles pinned at python3.8
other than that, this looks good to go

File: colin-api/Makefile (hunk 1 above)
Comment (argush3) [reply colin-api/Makefile:40]: ok, will revert

Comment (argush3) [reply colin-api/Makefile]: reverted

**Review by thorwolpert — APPROVED** [review]

**Pull request merged**